launcher/InstanceList.cpp

```diff
@@ -282,39 +255,79 @@ bool InstanceList::isGroupCollapsed(const QString& group)
     return m_collapsedGroups.contains(group);
 }
 
+bool InstanceList::trashInstance(const InstanceId& id)
+{
+    auto inst = getInstanceById(id);
+    if (!inst) {
+        qDebug() << "Cannot trash instance" << id << ". No such instance is present (deleted externally?).";
+        return false;
+    }
+
+    auto cachedGroupId = m_instanceGroupIndex[id];
+
+    if (m_instanceGroupIndex.remove(id)) {
+        saveGroupList();
+    }
```

Comment:
Turns out I was adding the group name to `m_trashHistory` after it was already deleted...

Now it works! 🎉

Reply:
(cc @DioEgizio)